### PR TITLE
Add [field:] and [property:] support for [RelayCommand] methods

### LINKED ISSUE #599: Allow custom field and property attributes for [RelayCommand]
### Overview

When trying to serialize an object that contains a `[RelayCommand]` the command property gets serialized by default, which could be undesirable.

We should be able to apply a custom property. to avoid serialization

### Usage example

```cs
[RelayCommand]
[property: Newtonsoft.Json.JsonIgnore]
private void DoWork() { }
```

### Breaking change?

No

### Alternatives

The alternative is to pass options during serialization.

### Help us help you

Yes, I'd like to be assigned to work on this item

### PR DESCRIPTION
**Pull request by Sergio0694**
**Closes #599**

@Youssef1313 do you mind also taking a quick look whenever you have time? Thank you! 😄

## PR Checklist

- [X] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [X] Based off latest main branch of toolkit
- [X] PR doesn't include merge commits (always rebase on top of our main, if needed)
- [X] Tested code with current [supported SDKs](../#supported)
- [X] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run _build/UpdateHeaders.bat_)
- [X] Contains **NO** breaking changes
- [X] Every new API (including internal ones) has full XML docs
- [X] Code follows all style conventions

### CODE REVIEW

File: src/CommunityToolkit.Mvvm.SourceGenerators/Diagnostics/Suppressors/RelayCommandAttributeWithFieldOrPropertyTargetDiagnosticSuppressor.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+            // Check that the target is effectively [field:] or [property:] over a method declaration, which is the case we're looking for
+            if (syntaxNode is AttributeTargetSpecifierSyntax { Parent.Parent: MethodDeclarationSyntax methodDeclaration, Identifier: SyntaxToken(SyntaxKind.FieldKeyword or SyntaxKind.PropertyKeyword) })
+            {
+                SemanticModel semanticModel = context.GetSemanticModel(syntaxNode.SyntaxTree);
+
+                // Get the method symbol from the first variable declaration
+                ISymbol? declaredSymbol = semanticModel.GetDeclaredSymbol(methodDeclaration, context.CancellationToken);
+
+                // Check if the field is using [ObservableProperty], in which case we should suppress the warning
```
Comment (Youssef1313) [inline src/CommunityToolkit.Mvvm.SourceGenerators/Diagnostics/Suppressors/RelayCommandAttributeWithFieldOrPropertyTargetDiagnosticSuppressor.cs:56]: Is this comment correct?

File: src/CommunityToolkit.Mvvm.SourceGenerators/Extensions/SymbolInfoExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    /// <param name="typeSymbol">The resulting attribute type symbol, if correctly resolved.</param>
+    /// <returns>Whether <paramref name="symbolInfo"/> is resolved to a symbol.</returns>
+    /// <remarks>
+    /// This can be used to ensure users haven't eg. spelled names incorrecty or missed a using directive. Normally, code would just
+    /// not compile if that was the case, but that doesn't apply for attributes using invalid targets. In that case, Roslyn will ignore
+    /// any errors, meaning the generator has to validate the type symbols are correctly resolved on its own.
+    /// </remarks>
+    public static bool TryGetAttributeTypeSymbol(this SymbolInfo symbolInfo, [NotNullWhen(true)] out INamedTypeSymbol? typeSymbol)
+    {
+        if ((symbolInfo.Symbol ?? symbolInfo.CandidateSymbols.SingleOrDefault()) is not ISymbol attributeSymbol ||
```
Comment (Youssef1313) [inline src/CommunityToolkit.Mvvm.SourceGenerators/Extensions/SymbolInfoExtensions.cs:29]: I'm worried about the possibility for an exception from `SingleOrDefault`? But I see this code existing prior to the PR

Comment (Sergio0694) [reply src/CommunityToolkit.Mvvm.SourceGenerators/Extensions/SymbolInfoExtensions.cs:29]: Oh right, that still throws if there's more than 1 item 🤦‍♂️
I'll change that, good catch ahah

**Review by Youssef1313 — APPROVED** [review]

**Review by Arlodotexe — APPROVED** [review]
No obvious issues. Looks good! 🚀

**Pull request merged**